src/workqueue.cpp

```diff
@@ -140,33 +142,40 @@ void WorkQueue::ThreadRun()
         assert(command != nullptr);
         assert(command->has_work());
 
-        // #TODO: Spawn jobs for command to unclog the queue.
-
-        WorkQueueResultPtr result = command->work();
+        // Submit work to thread pool
+        WorkQueueResultPtr result = ThreadPoolSingleton::get_workqueue_pool()
```

Comment:
I assume `.get()` means it waits for the task to finish? If yes, then this effectively is single threaded, just with a fancy task invoked, no?

Reply:
You're right - I've rewritten this to track and process pending tasks without blocking